src/Engine/Entity/Entity.cpp

```diff
@@ -21,6 +21,7 @@
 
 Entity::Entity(World* world, const std::string& name) : name ( name ) {
     this->world = world;
+    uid = rand()%1000000;
```

Comment:
Use c++11 random generators.

Also, this might produce identical UIDs, which can be a bug that is hard to track later.